### PR TITLE
Add nonceCondition

### PR DESCRIPTION
**Pull request by phanpak**

### CODE REVIEW

File: test/local/DSL/nonceCondition.test.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        { nonce: 256,        expectedError: false },
+
+        // Testing nonce just above the 256 boundary
+        { nonce: 257,        expectedError: false },
+
+        // Testing with a very large nonce to ensure handling of BigInt values
+        { nonce: BigInt(2)**BigInt(64), expectedError: false }
+    ];
+
+    edgeCases.forEach(testCase => {
```
Comment (phanpak) [inline test/local/DSL/nonceCondition.test.ts:61]: This style of testing is very popular in Go. I don't know if you guys are ok with it. I can split into their own tests if you prefer.

Comment (phanpak) [inline test/local/DSL/nonceCondition.test.ts:61]: and also a lot of these tests feel like testing nonceToBit which i'm not sure is useful. wdyt?

File: src/strategies/DSL/conditions/nonceCondition.ts [hunk 2]
```diff
@@ -1,11 +1,36 @@
...
+          }
+        }
+      ]
+    case STATE_NOT_USED:
+      return [
+        {
+          functionName: 'requireBitNotUsed',
+          params: {
+            bitmapIndex: bit.index,
+            bit: bit.value,
```
Comment (phanpak) [inline src/strategies/DSL/conditions/nonceCondition.ts:27]: @mikec do you mind checking what values we need for other parameters? 

* chainId
* signer

chainId is optional and signer is mandatory.

File: src/strategies/DSL/conditions/nonceCondition.ts [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
+  const bit = nonceToBit({nonce})
 
-  return []
+  const bitJson: BitJSON = { index: bit.index.toString(), value: bit.value.toString() }
+
+  switch (state) {
+    case STATE_USED:
+      return [
+        {
+          functionName: 'requireBitUsed',
```
Comment (mikec) [inline src/strategies/DSL/conditions/nonceCondition.ts:16]: I hadn't actually added primitive classes for the `requireBitUsed` and `requireBitNotUsed` functions yet, so I just did that and merged them to v5. You should rebase v5 branch so these are supported.

params expected are index and value. See how it's set up in tests here https://github.com/brinktrade/brink-sdk/blob/41c34a0b1bae8d5ad08314a7437de78fdda2af8b/test/local/Strategies.test.ts#L330

**Review by mikec — CHANGES_REQUESTED** [review]

File: src/strategies/DSL/conditions/nonceCondition.ts [hunk 4]
```diff
@@ -1,11 +1,32 @@
 import { PrimitiveArgs } from '@brinkninja/types'
-import { NonceConditionArgs } from '@brink-sdk'
+import { NonceConditionArgs, nonceToBit, BitJSON } from '@brink-sdk'
 
-function nonceCondition (def: NonceConditionArgs): PrimitiveArgs[] {
+const STATE_USED = 'USED'
```
[the remark this answers is not in the document]
Comment (phanpak) [reply src/strategies/DSL/conditions/nonceCondition.ts]: done

File: test/local/DSL/nonceCondition.test.ts (hunk 1 above)
Comment (phanpak) [reply test/local/DSL/nonceCondition.test.ts]: done

File: test/local/nonceToBit.test.ts [hunk 5]
```diff
@@ -19,9 +19,42 @@ describe('nonceToBit', () => {
     expect(() => nonceToBit({ nonce: 0 })).to.throw('Nonce should be greater than 0')
   })
 
-  it('should correctly convert large nonce', () => {
-    const bit = nonceToBit({ nonce: 1000 })
-    expect(bit.index).to.equal(3n)
-    expect(bit.value).to.equal(BigInt(2**231))
-  })
+  const edgeCases = [
```
Comment (mikec) [inline test/local/nonceToBit.test.ts:22]: awesome, thanks for adding the additional test cases!

**Review by mikec — APPROVED** [review]

**Review by mikec — APPROVED** [review]

**Pull request merged**